tutorials/kitties-part-01/5-step5.mdx

```diff
@@ -0,0 +1,319 @@
+# Kitties pallet: Dispatchables, events, and errors
+
+In the previous section, we laid down the foundations to manage the ownership of our 
+Kitties — even though they don't really exist yet! In this part of the tutorial, we'll be giving our pallet the 
+ability to create a Kitty using the storage items we've declared. Breaking things down, we're going to write:
+
+* `create_kitty`: a dispatchable or publicly callable function allowing an account to mint a Kitty.
+* `mint()`: a helper function that updates our pallet's storage items and performs error checks, called by 
+`create_kitty`.
+* pallet `Events`: using FRAME's `#[pallet::event]` attribute.
+
+At the end of this part, we'll check that everything compiles without error and call our `create_kitty` 
+extrinsic using the PolkadotJS Apps UI.
+
+### Public and private functions
+
+It is important to understand the pallet design decisions we'll be making around 
+coding up our Kitty pallet's minting and ownership management capabilities.
+
+We will be breaking up the "heavy lifting" logics into private helper functions. This improves code 
+readability and reusability.
+
+Lets paint the big picture of what combining dispatchables and helper functions looks like.
+
+`create_kitty` is a dispatchable function or extrinsic that:
+
+* Make sure the call is from a signed origin.
+* Generate unique DNA and Gender using a helper function.
+* States the generation as 0.
+* Calls a private mint() function.
+
+`mint()` is a private helper function that:
+
+* Check if the kitty does not already exist in our storage map.
+* Updates storage with the new Kitty ID (for all Kitties and for the owner's account).
+* Updates the new total Kitty count for storage and the new owner's account.
+* Deposits an `Event` to signal that a Kitty has succesfully been created.
+
+#### `create_kitty` dispatchable
+
+All pallet dispatchables live under the `#[pallet::call]` macro. All we [need to know](https://docs.substrate.io/reference/frame-macros/) here is that they're a 
+useful feature of FRAME that minimizes the code required to write for pallets to be properly integrated 
+in a Substrate chain's runtime.
+
+#### Weights
+
+Every dispatchable function must have an associated weight to it. Weights are an important part of 
+developing with Substrate as they provide safe-guards around the amount of computation to fit in a 
+block at execution time.
+
+Substrate's weighting system forces developers to think about the computational complexity each extrinsic 
+carries before it is called so that a node will account for it's worst case, avoiding lagging the 
+network with extrinsics that may take longer than the specified block time. Weights are also 
+intimately linked to the fee system for a signed extrinsic.
+
+For this application, and since we are developing it for learning purposes, we're going to default all weights to 0.
```

Comment:
Please add one line on where to find documentation about weight can be calculated.